content/docs/hooks-state.md

```diff
@@ -212,69 +213,69 @@ Let's now **recap what we learned line by line** and check our understanding.
 
 ```js
-  var fruitStateVariable = useState('banana'); // Returns a pair
-  var fruit = fruitStateVariable[0]; // First item in a pair
-  var setFruit = fruitStateVariable[1]; // Second item in a pair
+  var fruitStateVariable = useState('banana'); // Juftlik qaytariladi
+  var fruit = fruitStateVariable[0]; // Juftlikni birinchisi
+  var setFruit = fruitStateVariable[1]; // Juftlikni ikkinchisi
 ```
 
-When we declare a state variable with `useState`, it returns a pair — an array with two items. The first item is the current value, and the second is a function that lets us update it. Using `[0]` and `[1]` to access them is a bit confusing because they have a specific meaning. This is why we use array destructuring instead.
+`useState` orqali holat oʻzgaruvchisini yaratganimizda, u juftlik qaytaradi — ikki qismli massiv. Birinchi qismi joriy qiymat, ikkinchisida shu qiymatni oʻzgartiruvchi funksiya mavjud. Bu juftlik oʻziga xos ma'noga kasb eta oladi, `[0]` hamda `[1]` orqali ularni olish chalkashtiruvchi boʻlishi mumkin. Shuning uchun yozish uslubi massiv tarqatilishi turidan foydalanamiz.
 
->Note
+>Eslatma
 >
->You might be curious how React knows which component `useState` corresponds to since we're not passing anything like `this` back to React. We'll answer [this question](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) and many others in the FAQ section.
+>React qanday qilib `useState` qaysi komponentga tegishli ekanligini biladi deb qiziqayotgan boʻlishingiz mumkin, axir React'ga `this` kabi hech narsa berib yubormayapmiz. [Bu savolingizga](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) va boshqa koʻplab savollaringizga FAQ boʻlimida javob olishingiz mumkin.
 
-### Tip: Using Multiple State Variables {#tip-using-multiple-state-variables}
+### Maslahat: Bir qancha holat oʻzgaruvchilarini ishlatish {#tip-using-multiple-state-variables}
 
-Declaring state variables as a pair of `[something, setSomething]` is also handy because it lets us give *different* names to different state variables if we want to use more than one:
+Holat oʻzgaruvchilarini juftlik `[something, setSomething]` sifatida yaratish juda qulay, chunki bu bilan kerak boʻlsa *har xil* nomli, bittadan koʻp  holat oʻzgaruvchilarini yarata olamiz:
 
 ```js
 function ExampleWithManyStates() {
-  // Declare multiple state variables!
+  // Bir necha holat oʻzgaruvchilarini yaratamiz!
   const [age, setAge] = useState(42);
   const [fruit, setFruit] = useState('banana');
   const [todos, setTodos] = useState([{ text: 'Learn Hooks' }]);
 ```
 
-In the above component, we have `age`, `fruit`, and `todos` as local variables, and we can update them individually:
+Yuqoridagi komponentda, bizda `age`, `fruit` va `todos` ichki oʻzgaruvchilar bor va ularni birma bir yangilay olamiz:
 
 ```js
   function handleOrangeClick() {
-    // Similar to this.setState({ fruit: 'orange' })
+    // this.setState ga oʻxshash ({ fruit: 'orange' })
```

Comment:
**ga o'xshash** qismini `this.setState({ fruit: 'orange' })` dan keyin qo'yganingiz ma'qul menimcha.

Reply:
e'tiborsizlik tufayli qolib ketganga o'xshaydi, sinchikkovlik uchun rahmat